src/main/scala/com/codacy/parsers/implementation/CloverParser.scala

```diff
@@ -89,4 +107,34 @@ object CloverParser extends CoverageParser with XmlReportParser {
     node.attribute(attributeName).flatMap(_.headOption.map(_.text)).map(TextUtils.sanitiseFilename)
   }
 
+  /* Retrieves the first non-empty value in attribute with name @attributeName from @nodes
+   * and converts the value to a number
+   */
+  private def getFirstNonEmptyValueAsInt(nodes: Seq[Node], attributeName: String): Either[String, Int] = {
+    val attributeValues = nodes
+      .flatMap(_.attribute(attributeName).getOrElse(Seq.empty[Node]))
+
+    val firstNonEmptyValue = attributeValues
+      .map(_.text.trim)
+      .find(_.nonEmpty)
+
+    val result = firstNonEmptyValue.fold[Either[String, String]](
+      Left(s"Could not find attribute with name '$attributeName'")
+    )(Right(_))
+
+    result.flatMap { attributeValue =>
+      parseInt(attributeValue).left.flatMap(_ => Left(s"Value of attribute with name '$attributeName' is not a number"))
+    }
+  }
+
+  /* Parses string value into a number */
+  private def parseInt(value: String): Either[String, Int] = {
+    try {
```

Comment:
Codacy found an issue: [Consider using util.Try instead of a try-catch block](https://app.codacy.com/gh/codacy/coverage-parser/pullRequest?prid=6225255)